rust/datafusion/src/variable/user_defined.rs

```diff
@@ -0,0 +1,40 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+//! User defined variable provider
+
+use crate::error::Result;
+use crate::logicalplan::ScalarValue;
+use crate::variable::VarProvider;
+
+/// user defined variable
+pub struct UserDefinedVar {}
```

Comment:
This seems to be identical to `SystemVar`. I understand that we want to register separate variables for user-defined and system, but I don't think we need two separate implementations of the trait?

Reply:
Yes, I move it to test dir